src/pf-dropdown/pf-dropdown.component.js

```diff
@@ -0,0 +1,172 @@
+      if (event.target !== this && !this.contains(event.target)) {
+        this._clearDropdown();
+      }
+    });
+
+    document.addEventListener('keydown', (event) => {
+      if (/input|textarea/.test(event.target.tagName)) {
+        return;
+      }
+      if (this._disabled) {
+        return;
+      }
+      let keycode = event.keyCode ? event.keyCode : event.which;
+      let active = /\bopen/.test(this._button.parentNode.className);
+
+      //check if dropdown is open and keys other than 38,40,27,13 are not being used
+      if (active && /(38|40|27|13)/.test(keycode)) {
+        this._keyHandler(event, keycode);
+      }
+    });
+  }
+
+  /**
+   * Called when element's attribute value has changed
+   *
+   * @param {string} attrName The attribute name that has changed
+   * @param {string} oldValue The old attribute value
+   * @param {string} newValue The new attribute value
+   */
+  attributeChangedCallback(attrName, oldValue, newValue) {
+
+  }
+
+  /**
+   * Called when an instance of the element is created
+   */
+  createdCallback() {
+
+  }
+
+  /**
+   *Toggle the dropdown
+   */
+  toggle() {
+    this._showDropdown();
+  }
+
+  /**
+   * Open the dropdown
+   *
+   */
+  _showDropdown() {
+    if (/\bdisabled/.test(this._button.className)) {
+      return;
+    }
+    let active = this._button.parentNode.classList.contains('open');
+    if (!active) {
+      this._detectTouch();
+      this.dispatchEvent(new CustomEvent('show.bs.dropdown', {}));
+      this._button.focus();
+      this._button.setAttribute('aria-expanded', 'true');
+      this._button.parentNode.classList.toggle('open');
+      this.dispatchEvent(new CustomEvent('shown.bs.dropdown', {}));
+    }
+    if (active) {
+      this._clearDropdown();
+    }
+  }
+  /**
+   * Close the dropdown
+   *
+   */
+  _clearDropdown() {
+    let backdrop = this.querySelector('.dropdown-backdrop');
+    if (backdrop) {
+      backdrop.parentNode.removeChild(backdrop);
+    }
+    this.dispatchEvent(new CustomEvent('hide.bs.dropdown', {}));
+    this._button.setAttribute('aria-expanded', 'false');
+    this._button.parentNode.classList.remove('open');
+    this.dispatchEvent(new CustomEvent('hidden.bs.dropdown', {}));
+  }
+
+  /**
+   * Support for phone browser
+   *
+   */
+  _detectTouch() {
+    if ('ontouchstart' in document.documentElement) {
+      let div = document.createElement('div');
+      div.classList.add('dropdown-backdrop');
+      this.insertBefore(div, this.querySelector('.dropdown-menu'));
+      div.addEventListener('click', this._clearDropdown());
+    }
+  }
+
+  /**
+   *Support for accessibility
+   *
+   * @param {Event} event
+   */
+  _keyHandler(event, keycode) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (keycode === 27) {
+      this._clearDropdown();
+      this._button.focus();
+    }
+    if (keycode === 13) {
+      event.target.dispatchEvent(new MouseEvent('click'));
```

Comment:
Why is the original event canceled, only to dispatch a new event here?

Instead of cancelling all events up front, it may be best to move that code to the if statements which need it. Or, move it to  a function and call it as needed?